lib/components/text_box_component.dart

```diff
@@ -103,39 +105,43 @@ class TextBoxComponent extends PositionComponent with Resizable {
     return _lines.length - 1;
   }
 
-  double _withMargins(double size) => size + 2 * _boxConfig.margin;
-
   @override
-  double get width => currentWidth;
-
-  @override
-  double get height => currentHeight;
-
-  double get totalWidth => _withMargins(_maxLineWidth);
-
-  double get totalHeight => _withMargins(_lineHeight * _lines.length);
+  Vector2 get size => Vector2(width, height);
 
   double getLineWidth(String line, int charCount) {
-    return _withMargins(_config
+    return _config
         .toTextPainter(line.substring(0, math.min(charCount, line.length)))
-        .width);
+        .width;
   }
 
-  double get currentWidth {
-    int i = 0;
-    int totalCharCount = 0;
-    final int _currentChar = currentChar;
-    final int _currentLine = currentLine;
-    return _lines.sublist(0, _currentLine + 1).map((line) {
-      final int charCount =
-          (i < _currentLine) ? line.length : (_currentChar - totalCharCount);
-      totalCharCount += line.length;
-      i++;
-      return getLineWidth(line, charCount);
-    }).reduce(math.max);
+  @override
+  double get width {
+    if (_boxConfig.growingBox) {
```

Comment:
I think we should cache this? I expect that component.width to be O(0)

Reply:
It depends if it is a growing box or not, but yeah, it can be cacheable even if it is not. I'll look into it.